core/src/main/scala/com/mesosphere/usi/core/Scheduler.scala

```diff
@@ -130,4 +147,25 @@ object Scheduler {
       }
     }
   }
+
+  private def persistenceFlow(
+      podRecordRepository: PodRecordRepository): Flow[SchedulerEvents, SchedulerEvents, NotUsed] =
+    Flow[SchedulerEvents].mapAsync(1) { events =>
+      import CallerThreadExecutionContext.context
+      Future
+        .sequence(events.stateEvents.map {
```

Comment:
In the current flow of things:

```
[SchedulerEvents] ------>        [Persistence]    ----------------> [StateOutput, MesosCalls]
emitted by 				 process ALL the StateEvents
Scheduler logic			 in single SchedulerEvents
						 object and THEN emits the 
						 object downstream.
						 backpressures the upstream 
						 until ALL the StateEvents
						 in SchedulerEvents are processed.
						 We can use a stream here but i 
					     am not sure what advantage it would
					     add. We are using `.mapAsync` with 				
						 parallelism 1 to process each
						 SchedulerEvents object - one at a time
						 with backpressure.
```

Currently the persistence layer emits the **exact** `SchedulerEvents` object that it receives, are you perhaps inferring that we don't need to do this and emit a "split-down" SchedulerEvents based on the stream processing speed ?

Reply:
In some place in Marathon we used Akka streams instead of `Future.sequence` to avoid too many futures. Also, there is not backpressure for loading.

Reply:
So, I think @zen-dog can help. I thought we had something like a `Flow[StoreOp, Try]` So that one could map the updates.

```
val persitenceFlow: Flow[SchedulerEvents, SchedulerEvents] = {
  Flow[SchedulerEvents]
    .flatMapConcat(_.stateEvents)
    .map {
       case ... => UpdateOp | DeleteOp
    }
    .via(repository.flow)
    ...
}
```

This might be a little to complicated though it also omits then mapping back to `SchedulerEvents`.